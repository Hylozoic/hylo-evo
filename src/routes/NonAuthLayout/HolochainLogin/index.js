import component from './HolochainLogin'
import connector from './HolochainLogin.holochain.connector'
export default connector(component)
