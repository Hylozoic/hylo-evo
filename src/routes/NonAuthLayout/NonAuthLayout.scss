$third-party-icon-size: 14px;

.background {
  width: 100vw;
  height: 100vh;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}


.particlesBackgroundWrapper {
  position: absolute;
  display: block;
  top: 0;
  left: 0;
  z-index: -1000;
  height: 100vh;
  width: 100vw;
  line-height: 100vh;
}

.topRow {
  position: absolute;
  top: 0px;
  width: 100%;
  display: flex;
  justify-content: space-between;
  padding: 20px;
}

.form {
  background-color: white;
  box-shadow: 0px 0px 6px 0px $color-black-10;
  padding: 50px 40px 40px;
  border-radius: 8px;
  width: 420px;
}

.below-container {
  margin-top: $space-2x;
  a {
    color: white !important;
    cursor: pointer;
  }
}

.signupButton {
  background-color: $color-white;
}

.logo {
  /* !empty */
}

.green-text {
  color: $color-caribbean-green
}

.error {
  color: $color-amaranth;
  background-color: transparentize($color-amaranth, 0.9);
  border: 1px solid $color-amaranth;
  border-radius: 4px;
  padding: 9px 16px;
  margin-bottom: 20px;
}
