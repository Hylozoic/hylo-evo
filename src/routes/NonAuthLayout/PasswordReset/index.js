import component from './PasswordReset'
import connector from './PasswordReset.connector'
export default connector(component)
