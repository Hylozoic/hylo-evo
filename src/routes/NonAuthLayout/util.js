import React from 'react'
import { uniq } from 'lodash'
import './NonAuthLayout.scss'

export function formatError (error, action) {
  if (!error) return

  const noPasswordMatch = error.match(/password account not found. available: \[(.*)\]/)
  if (noPasswordMatch) {
    var options = uniq(noPasswordMatch[1].split(',')
    .map(option => ({
      'google': 'Google',
      'google-token': 'Google',
      'facebook': 'Facebook',
      'facebook-token': 'Facebook',
      'linkedin': 'LinkedIn',
      'linkedin-token': 'LinkedIn'
    }[option])))

    return <div styleName='error'>
      Your account has no password set. <a href='/reset-password'>Set your password here.</a>
      {options[0] && <span><br />Or log in with {options.join(' or ')}.</span>}
    </div>
  }

  var text
  switch (error) {
    case 'no user':
      text = `${action} was canceled or no user data was found.`
      break
    case 'no email':
      text = 'The user data did not include an email address.'
      break
    default:
      text = error
  }

  return <div styleName='error'>{text}</div>
}
