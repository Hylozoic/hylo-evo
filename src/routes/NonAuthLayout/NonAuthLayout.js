import React from 'react'
import { Route, Link } from 'react-router-dom'
import Particles from 'react-particles-js'
import particlesjsConfig from './particlesjsConfig'
import Button from 'components/Button'
import Login from './Login'
import Signup from './Signup'
import PasswordReset from 'routes/NonAuthLayout/PasswordReset'
import './NonAuthLayout.scss'

export default class NonAuthLayout extends React.Component {
  constructor (props) {
    super(props)
    this.state = {}
  }

  render () {
    const particlesStyle = {
      position: 'fixed',
      top: 0,
      left: 0,
      width: '100%',
      height: '100%'
    }
    return <div styleName='background'>
      <div styleName='particlesBackgroundWrapper'>
        <Particles params={particlesjsConfig} style={particlesStyle} />
      </div>
      <div styleName='topRow'>
        <a href='/'>
          <img styleName='logo' src='assets/hylo.svg' />
        </a>
        <Route path='/login' component={() =>
          <Link to='/signup'>
            <Button styleName='signupButton' color='green-white-green-border'>Sign Up</Button>
          </Link>
        } />
      <Route path='/reset-password' component={() =>
          <Link to='/login'>
            <Button styleName='signupButton' color='green-white-green-border'>Log In</Button>
          </Link>
        } />
      </div>

      <Route path='/login' component={() =>
        <Login {...this.props} styleName='form' />
      } />

      <Route path='/signup' component={() =>
        <Signup {...this.props} styleName='form' />
      } />

      <Route path='/reset-password' component={() =>
        <PasswordReset {...this.props} styleName='form' />
      } />

      <div>
        <p styleName='below-container'>
          <Route path='/signup' component={() =>
            <Link to='/login'>
              Already have an account? <span styleName='green-text'>Sign in</span>
            </Link>
          } />
        </p>
      </div>
    </div>
  }
}
