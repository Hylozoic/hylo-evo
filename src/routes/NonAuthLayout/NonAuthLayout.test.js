import NonAuthLayout from './NonAuthLayout'
import { shallow } from 'enzyme'
import React from 'react'

describe('NonAuthLayout', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<NonAuthLayout />)
    expect(wrapper).toMatchSnapshot()
  })
})
