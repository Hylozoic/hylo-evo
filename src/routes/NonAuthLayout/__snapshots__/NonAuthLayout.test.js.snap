// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NonAuthLayout renders correctly 1`] = `
<div
  data-stylename="background"
>
  <div
    data-stylename="particlesBackgroundWrapper"
  >
    <t
      height="100%"
      params={
        Object {
          "interactivity": Object {
            "detect_on": "canvas",
            "events": Object {
              "onclick": Object {
                "enable": false,
                "mode": "push",
              },
              "onhover": Object {
                "enable": false,
                "mode": "repulse",
              },
              "resize": true,
            },
            "modes": Object {
              "bubble": Object {
                "distance": 400,
                "duration": 2,
                "opacity": 8,
                "size": 40,
                "speed": 3,
              },
              "grab": Object {
                "distance": 400,
                "line_linked": Object {
                  "opacity": 1,
                },
              },
              "push": Object {
                "particles_nb": 4,
              },
              "remove": Object {
                "particles_nb": 2,
              },
              "repulse": Object {
                "distance": 200,
                "duration": 0.4,
              },
            },
          },
          "particles": Object {
            "color": Object {
              "value": "#0dc39f",
            },
            "line_linked": Object {
              "color": "#0dc39f",
              "distance": 150,
              "enable": true,
              "opacity": 0.4,
              "width": 1,
            },
            "move": Object {
              "attract": Object {
                "enable": true,
                "rotateX": 600,
                "rotateY": 1200,
              },
              "bounce": false,
              "direction": "none",
              "enable": true,
              "out_mode": "out",
              "random": true,
              "speed": 1,
              "straight": false,
            },
            "number": Object {
              "density": Object {
                "enable": true,
                "value_area": 800,
              },
              "value": 32,
            },
            "opacity": Object {
              "anim": Object {
                "enable": false,
                "opacity_min": 0.1,
                "speed": 1,
                "sync": false,
              },
              "random": true,
              "value": 0.5,
            },
            "shape": Object {
              "image": Object {
                "height": 100,
                "src": "img/github.svg",
                "width": 100,
              },
              "polygon": Object {
                "nb_sides": 5,
              },
              "stroke": Object {
                "color": "#000000",
                "width": 0,
              },
              "type": "circle",
            },
            "size": Object {
              "anim": Object {
                "enable": false,
                "size_min": 0.1,
                "speed": 40,
                "sync": false,
              },
              "random": true,
              "value": 4.008530152163807,
            },
          },
          "retina_detect": true,
        }
      }
      style={
        Object {
          "height": "100%",
          "left": 0,
          "position": "fixed",
          "top": 0,
          "width": "100%",
        }
      }
      width="100%"
    />
  </div>
  <div
    data-stylename="topRow"
  >
    <a
      href="/"
    >
      <img
        alt="Hylo logo"
        data-stylename="logo"
        src="assets/hylo.svg"
      />
    </a>
    <Route
      component={[Function]}
      path="/login"
    />
    <Route
      component={[Function]}
      path="/reset-password"
    />
  </div>
  <Route
    component={[Function]}
    path="/login"
  />
  <Route
    component={[Function]}
    path="/signup"
  />
  <Route
    component={[Function]}
    path="/reset-password"
  />
  <div>
    <p
      data-stylename="below-container"
    >
      <Route
        component={[Function]}
        path="/signup"
      />
    </p>
  </div>
</div>
`;
